cli/src/pcluster/schemas/common_schema.py

```diff
@@ -111,3 +112,29 @@ class TagSchema(BaseSchema):
     def make_resource(self, data, **kwargs):
         """Generate resource."""
         return BaseTag(**data)
+
+
+class CookbookSchema(BaseSchema):
+    """Represent the schema of the ImageBuilder chef cookbook."""
+
+    chef_cookbook = fields.Str()
+    extra_chef_attributes = fields.Str()
+
+    @post_load()
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return Cookbook(**data)
+
+    @validates("extra_chef_attributes")
+    def validate_extra_chef_attributes(self, value):
+        """Validate json."""
+        if value and not validate_json_format(value):
+            raise ValidationError(message="'{0}' is invalid".format(value))
+
+
+class BaseDevSettingsSchema(BaseSchema):
+    """Represent the common schema of Dev Setting for ImageBuilder and Cluster."""
+
+    cookbook = fields.Nested(CookbookSchema)
+    node_package = fields.Str()
+    aws_batch_cli_package = fields.Str(data_key="AWSBatchCLIPackage")
```

Comment:
We should align the convention, so use AwsBatchCliPackage

Reply:
Done